tests/include/test/macros.h

```diff
@@ -73,7 +83,7 @@
 #define TEST_EQUAL(expr1, expr2)                                      \
     do {                                                                \
         if (!mbedtls_test_equal( #expr1 " == " #expr2, __LINE__, __FILE__, \
-                                 expr1, expr2))                      \
+                                 (unsigned long long) (expr1), (unsigned long long) (expr2)))                      \
```

Comment:
Please don't combine unrelated changes in the same commit.

Why this change anyway?

Reply:
It was needed for another PR as IAR was complaining about implicit type conversion and mbedtls_test_equal was defined to have arguments of type unsigned long long.

Reply:
> IAR was complaining about implicit type conversion and mbedtls_test_equal was defined to have arguments of type unsigned long long.

That's a valid argument, but it belongs in its own commit.

Though I don't like that it causes `TEST_EQUAL(p1, p2)` to compile when `p1` and `p2` are pointers. It's not catastrophic since on most platforms it will compare the pointers for equality, but it can have both false positives and false negatives on “exotic” architectures, especially architectures with rich pointers. Can we find a way to say that converting signed integer types are ok, but pointers are not ok?

Reply:
I'll have a look into that in #7935